mongocrypt/src/binary.rs

```diff
@@ -0,0 +1,78 @@
+use std::borrow::Borrow;
+
+use mongocrypt_sys as sys;
+
+use crate::convert::binary_bytes;
+use crate::error::Result;
+use crate::native::OwnedPtr;
+
+pub(crate) struct Binary {
+    inner: OwnedPtr<sys::mongocrypt_binary_t>,
+}
+
+impl Binary {
+    pub(crate) fn new() -> Self {
+        Self {
+            inner: OwnedPtr::new(
+                unsafe { sys::mongocrypt_binary_new() },
+                sys::mongocrypt_binary_destroy,
+            ),
+        }
+    }
+
+    fn new_from(inner: *mut sys::mongocrypt_binary_t) -> Self {
+        Self {
+            inner: OwnedPtr::new(inner, sys::mongocrypt_binary_destroy),
+        }
+    }
+
+    pub(crate) fn native(&self) -> &*mut sys::mongocrypt_binary_t {
+        self.inner.borrow()
+    }
+
+    pub(crate) unsafe fn bytes<'a>(&self) -> Result<&'a [u8]> {
```

Comment:
Can we add a comment indicating why this is unsafe/dangerous? i.e. point out that the lifetime of the returned bytes is caller specified and is not tied to `&self`.

Reply:
Done.